packages/components/src/color-picker/index.js

```diff
@@ -106,140 +113,137 @@ const dataToColors = ( oldColors, { source, valueKey, value } ) => {
+		draftRgb: initialColor.rgb,
+		draftHsl: initialColor.hsl,
+	} );
+
+	const debouncedOnChangeComplete = useCallback(
+		useDebounce( onChangeComplete, 100 ),
+		[ onChangeComplete ]
+	);
+
+	const { hex, hsl, hsv, rgb, draftHex, draftHsl, draftRgb } = colors;
+
+	const classes = classnames( className, {
+		'components-color-picker': true,
+		'is-alpha-disabled': disableAlpha,
+		'is-alpha-enabled': ! disableAlpha,
+	} );
+
+	function commitValues( data ) {
 		if ( isValidColor( data ) ) {
-			const colors = colorToState( data, data.h || oldHue );
-			this.setState(
-				{
-					...colors,
-					draftHex: toLowerCase( colors.hex ),
-					draftHsl: colors.hsl,
-					draftRgb: colors.rgb,
-				},
-				debounce( partial( onChangeComplete, colors ), 100 )
-			);
+			const newColors = colorToState( data, data.h || oldHue );
+			setColors( {
+				...newColors,
+				draftHex: toLowerCase( newColors.hex ),
+				draftHsl: newColors.hsl,
+				draftRgb: newColors.rgb,
+			} );
+
+			debouncedOnChangeComplete( newColors );
 		}
 	}
 
-	resetDraftValues() {
-		this.setState( {
-			draftHex: this.state.hex,
-			draftHsl: this.state.hsl,
-			draftRgb: this.state.rgb,
+	function resetDraftValues() {
+		setColors( {
+			...colors,
+			draftHex: hex,
+			draftHsl: hsl,
+			draftRgb: rgb,
 		} );
 	}
 
-	setDraftValues( data ) {
+	function setDraftValues( data ) {
 		switch ( data.source ) {
 			case 'hex':
-				this.setState( { draftHex: toLowerCase( data.hex ) } );
+				setColors( {
+					...colors,
+					draftHex: toLowerCase( data.hex ),
+				} );
 				break;
 			case 'rgb':
-				this.setState( { draftRgb: data } );
+				setColors( {
+					...colors,
+					draftRgb: data,
+				} );
 				break;
 			case 'hsl':
-				this.setState( { draftHsl: data } );
+				setColors( {
+					...colors,
+					draftHsl: data,
+				} );
 				break;
 		}
 	}
 
-	handleInputChange( data ) {
+	function handleInputChange( data ) {
 		switch ( data.state ) {
 			case 'reset':
-				this.resetDraftValues();
+				resetDraftValues();
 				break;
 			case 'commit':
-				const colors = dataToColors( this.state, data );
-				if ( ! isValueEmpty( colors ) ) {
-					this.commitValues( colors );
+				const newColors = dataToColors( colors, data );
+				if ( ! isValueEmpty( newColors ) ) {
+					commitValues( newColors );
 				}
 				break;
 			case 'draft':
-				this.setDraftValues( dataToColors( this.state, data ) );
+				setDraftValues( dataToColors( colors, data ) );
 				break;
 		}
 	}
 
-	render() {
-		const { className, disableAlpha } = this.props;
-		const {
-			color,
-			hsl,
-			hsv,
-			rgb,
-			draftHex,
-			draftHsl,
-			draftRgb,
-		} = this.state;
-		const classes = classnames( className, {
-			'components-color-picker': true,
-			'is-alpha-disabled': disableAlpha,
-			'is-alpha-enabled': ! disableAlpha,
-		} );
+	return (
+		<div className={ classes }>
+			<div className="components-color-picker__saturation">
+				<Saturation hsl={ hsl } hsv={ hsv } onChange={ commitValues } />
```

Comment:
`commitValues` is a new function on every render which means the `useCallback` won't work.

Reply:
What if i move `useThrottle` from `Saturation` to `ColorPicker` ?  by putting `commitValues` function inside `useThrottle` and `useCallback` hooks in `ColorPicker` level, so the throttled callback will be passed to `Saturation`.

Reply:
I'm not sure if it will matter much. As long as there is a dependency on `onChangeComplete`, `useCallback` will just return a new function on every call too. This may be alright if the function is not re-rendering all the time. I'll see if I can test this PR tomorrow.

Reply:
Thank you @ellatrix, i've tested with a higher delay in `useDebounce` and `useThrottle` and I they are working well for me. and re-render of `ColorPicker` happen on every UI interraction.